### ISSUE TITLE
Stabilize filter missing if GPU Processing enabled

### ISSUE THREAD

**Issue by qubodup** [mltframework/shotcut#559]
In [a YouTube comment thread](https://www.youtube.com/watch?v=PpYPsgnOxcY&lc=Ugyg6qJn40diwc2KkwZ4AaABAg), it was pointed out that "Stabilize" is not visible in various versions (I tested in 18.05.08) if GPU Processing is enabled.

Situation 1:

Reproduction steps:
1. Enable GPU Processing
2. Add a clip, try to apply Filter "Stabilize".

Expected result:
3. a "Stabilize" filter is available.

Actual result:
3. a "Stabilize" filter is not available.

Situation 2:
Reproduction steps:
1. Add a clip, apply Filter "Stabilize".
2. Save project.
3. Enable GPU Processing.

Expected result:
4. Stabilize would be removed or can still be added to other clips

Actual result:
4. a "Stabilize" filter is not available but it is kept in the clip to which it already was applied.

Notes:
1. Rendering was not tested in GPU Processing mode.
2. If Stabilize can work in GPU Processing mode, then it should be allowed to be added while in GPU Processing mode
3. If Stabilize cannot work in GPU Processing mode, then - in GPU Processing mode - it should be
a) removed or
b) disabled and not allowed to be enabled (ideally with a hover or in-ui info clarifying why)

Urgency:
I don't think there is urgency to this ticket, as I have never seen complaints about this. Only after a video tutorial was published, I noticed people having issue with this.

Off-topic:
In general, I would also recommend showing all filters in both GPU Processing mode and non-GPU Processing mode but the ones that cannot work in the current mode should be grayed out and non-clickable with a note "(Only available in GPU Processing mode)" or "(Not available in GPU Processing mode)" - ideally in the filter title and mouse hover popup. Unfortunately this might be more work than it's worth, especially with filters being constantly changed (I imagine).

**Comment by bmatherly** [comment]
Situation 2 is fixed by this pull request:
https://github.com/mltframework/shotcut/pull/551/files#diff-3338ecdc765ec0a1d0548ce990a57993R360

In the new code, Shotcut will not open the file with the Stabalize effect in GPU mode. It will prompt the user to disable GPU mode.

It also adds some text to clarify that some effects will not work:
https://github.com/mltframework/shotcut/pull/551/files#diff-585cfecf0ce15ac75be3038e77e65611R2893

Open to comments on the wording. Trying to make it clearer without making it too verbose.

**Comment by ddennedy** [comment]
Situation 1 is expected, and I disagree with the off-topic.

Re Situation 2, I do not want to remove filters upon switching modes. Brian's change addresses because it changes behavior to force a restart if you agree to the confirmation dialog, and does not apply the change otherwise. Then, upon reloading the project in GPU mode, if stabilize is included, it alerts the user and asks them if they want to change modes and restart.

In summary, Brian's pull request is the fix for this.

**Comment by kikislater** [comment]
In Archlinux : GPU option disappear so Stabilize filter is not available ! 
Any thought ?

**Comment by ddennedy** [comment]
Your build was probably missing the vid.stab library when MLT was configured for build. It is not missing because GPU is off or hidden.

**Comment by kikislater** [comment]
So what happen to GPU option for now ? It's not really clear for me. Ok I removed my home preference and it solved problem ! Thank you.
Do I have to open a ticket if I want to know how I could find frei0r filter like ndvi inside shotcut ?

**Comment by ddennedy** [comment]
ndvi is not available yet. Shotcut, by design, does not automatically expose all effects with a generic UI like other tools. ndvi will not be considered until I add categories and descriptions to the filters menu.

**Comment by kikislater** [comment]
Ok thank you for your answer. I asked because this morning I discover your "fork" : Olive and found that there is ndvi inside. 
So I have to custom GUI with Qt (qml and ui)? Is there a shared resources between users with these custom Ui files : I mean not how to do (I found here : https://www.shotcut.org/notes/make-plugins/ ) but a kind of repository where users have already do it (not only ndvi).
